lib/fastlane/plugin/revenuecat_internal/helper/versioning_helper.rb

```diff
@@ -161,18 +161,91 @@ def self.detect_bump_type(version_name_a, version_name_b)
         end
       end
 
+      def self.validate_input_if_appending_phc_version?(append_hybrid_common_version, include_prereleases, hybrid_common_version)
+        if append_hybrid_common_version
+          UI.user_error!("Appending the PHC version to prerelease versions violates SemVer.") if include_prereleases
+          UI.user_error!("Cannot append a nil PHC version.") if hybrid_common_version.nil?
+          UI.user_error!("Cannot append a blank PHC version.") if hybrid_common_version.strip.empty?
+        end
+      end
+
+      def self.validate_new_version_if_appending_phc_version?(append_hybrid_common_version, new_version_number, hybrid_common_version)
+        if append_hybrid_common_version
+          if new_version_number.include?("-")
+            UI.user_error!("Appending the PHC version to prerelease versions violates SemVer.")
+          end
+          if new_version_number.include?("+") && new_version_number.partition("+").last != hybrid_common_version
+            UI.user_error!(
+              "Asked to append PHC version (+#{hybrid_common_version}), " \
+              "but the provided version (#{new_version_number}) already has metadata " \
+              "(+#{new_version_number.partition('+').last})."
+            )
+          end
+        end
+      end
+
+      def self.append_phc_version_if_necessary(append_hybrid_common_version, include_prereleases, hybrid_common_version, new_version_number)
+        if should_ask_to_append_phc_version?(append_hybrid_common_version, include_prereleases, hybrid_common_version, new_version_number)
+          if UI.confirm("Would you like to append the PHC version (+#{hybrid_common_version})?")
+            return "#{new_version_number}+#{hybrid_common_version}"
+          end
+        elsif should_append_phc_version?(append_hybrid_common_version, include_prereleases, hybrid_common_version, new_version_number)
+          UI.important(
+            "Appending PHC version (+#{hybrid_common_version}) to new version (#{new_version_number}), as instructed."
+          )
+          return "#{new_version_number}+#{hybrid_common_version}"
+        end
+        return new_version_number
+      end
+
+      private_class_method def self.should_ask_to_append_phc_version?(append_hybrid_common_version, include_prereleases, hybrid_common_version, new_version_number)
```

Comment:
Hmm I'm thinking what the use case for this one is? This will only be asked if done locally right? In that case, I think we should just try to imitate what we do in CI? Maybe I'm missing a case we would want something different locally though...

Reply:
The idea here was to minimize the chance of errors when releasing locally (e.g. majors). Without this you'd need to remember to append the PHC version when releasing purchases-kmp. Now, if you input a version number without the build metadata, it asks if you want to append the PHC version as build metadata. 

I guess an alternative would be to hard code a list of repos for which we always want to append the PHC version?

Reply:
Hmm but the repos using PHC as metadata would already define the `append_hybrid_common_version` parameter to `true` right? So I think it's ok to assume if you're bumping the version in a repo where that's defined without passing the metadata, just append the PHC version. Wdyt?

Reply:
I see what you mean. I think we can make that work with a slight adjustment. `append_hybrid_common_version` should be something like `append_hybrid_common_version_if_next_version_not_prerelease` (and behave in that way too). 
 
The reason why I'm thinking that is that we could call `bump` (manually) without specifying a version (yet). If we then input a prerelease version (when the CLI asks for it), and `append_hybrid_common_version` is hardcoded to true in the repo, that would be an invalid state. (We currently throw an error.) However we should be able to release prerelease versions via the CLI of course.

Hope it's somewhat clear. What do you think?

Reply:
Ahhh yes I understand the issue now. And yeah I think your proposed solution should work 👍

Reply:
This is now done! We never ask the user anymore. Repos that need to append the PHC version will have `append_phc_version_if_next_version_is_not_prerelease` hardcoded to true. 

Had to adjust some tests for this, and could remove some code. If `append_phc_version_if_next_version_is_not_prerelease` is true, but the logic decides _not_ to append, the reason for that decision is printed to std out. (See `VersioningHelper.should_append_phc_version?()`.)